### DISCUSSION TITLE
Is Sonarcloud included?

### DISCUSSION THREAD

**Discussion by conbrad** [bcgov/quickstart-openshift#1149] [Q&A] [answered]
Is sonarcloud code scanning included OOTB? I saw in the README under unit tests it can be optionally used, but is that for code coverage?

**Comment by DerekRoberts** [comment, marked as answer]
It's included out of the box, but gets skipped if there's no SONAR_TOKEN.  If you need to ask, then I need to update the docs!  ...poop...

**Reply by conbrad** [reply]
Cool tbh I only skimmed the docs so it may just be me. Maybe I need to start a new discussion on this but do you know of a way to manage updated sonarcloud rules? We've run into issues where our main branch fails as soon as they update some rules, and I haven't found a way to triage and accept new rules or some other sort of workflow. Thoughts?

https://docs.sonarcloud.io/digging-deeper/rules/#what-might-change-after-an-update

**Reply by DerekRoberts** [reply]
We use the same workflow for PRs and main merge, so the window for rule changes to sneak in is very small.  Anything that does would be handled in a stop-the-pipeline-type PR.

Is there a link or explanation that'd make your block a little easier to follow?  Are these kicking up in scheduled workflows?

**Reply by conbrad** [reply]
> Is there a link or explanation that'd make your block a little easier to follow? Are these kicking up in scheduled workflows?

The best link that explains the behaviour I've found is: https://docs.sonarcloud.io/digging-deeper/rules/#what-might-change-after-an-update

The closest thing I found to a changelog for Sonarcloud rules is the "Available since" metadata for a rule: https://docs.sonarcloud.io/digging-deeper/rulesIt's defined as: "the date when a rule was first added on SonarCloud. This is useful to list all the new rules since the last upgrade of a plugin for instance."

Basically, sonarcloud does silent rule updates, so it breaks repeatability in running workflows if you rely on sonarcloud. Just a small frustration really. Just wondering if you've come across it and/or know if it's something that can be configured.

**Reply by DerekRoberts** [reply]
That's infuriating!  It breaks all our patterns for repeatability and consistency.  :'(

**Reply by conbrad** [reply]
Looks like they suggest using their non-managed sonarqube as a workaround: https://community.sonarsource.com/t/sonarcloud-rules-versioning/31688/2

**Reply by DerekRoberts** [reply]
So solve it with money?  :/

**Reply by franTarkenton** [reply]
It would be SO nice if sonarcloud was smart enough to load the rules that you want to use to evaluate the code base from the repo itself.  Even better would be to load rules that are defined in your linter.  As I recall the only way to disable a rule is to add a #nosonar tag to the line in question.

**Reply by conbrad** [reply]
> So solve it with money? :/

I edited my comment to change from "managed" to "unmanaged"(my bad!) I think it's basically run it yourself if you want to manage rule changes.
